src/parsers/PnmlParser.js

```diff
@@ -109,8 +143,15 @@ export default class PnmlParser {
         }
     }
 
-    getName (node) {
-        return node.firstElementChild.firstChild.nodeValue;
+    sortElements () {
+        const sortOrder = {
+            'shape': 0,
+            'label': 1,
+            'connection': 2,
```

Comment:
Why there have to be a sort order?

Reply:
Basically to make sure that all references are there before they are needed (e.g. label.parent, connection.target, etc.).